modules/ctl/src/main/scala/low/git.scala

```diff
@@ -0,0 +1,58 @@
+package gem.ctl
+package low
+
+import scalaz._, Scalaz._
+import scala.util.matching.Regex
+
+import gem.ctl.free.ctl._
+import io._
+
+/** Low-level constructors for `CtlIO` operations related to git. */
+object git {
+
+  case class Commit(hash: String)
+  object Commit {
+    implicit val OrderCommit: Order[Commit] = Order.orderBy(_.hash)
+  }
+
+  case class Tag(tag: String)
+
+  // Yields most recent tag matching the given pattern, if any
+  def mostRecentTag(pattern: Regex): CtlIO[Option[Tag]] =
+    shell("git", "tag").require {
```

Comment:
This is a hard problem but how do we go accounting for differences on the command versions if e.g. different users have slightly different git configs that result in a different output?

Could/Should we have used a git library if such thing exists?